src/FirmwarePlugin/PX4/PX4FirmwarePlugin.cc

```diff
@@ -174,7 +174,7 @@ bool PX4FirmwarePlugin::setFlightMode(const QString& flightMode, uint8_t* base_m
 
 int PX4FirmwarePlugin::manualControlReservedButtonCount(void)
 {
-    return 8;   // 8 buttons reserved for rc switch simulation
+    return 0;   // 0 buttons reserved for rc switch simulation
```

Comment:
> We have flight time on this and have not had issues with inadvertently getting an RC switch action from button presses.

It will turn buttons in rc channels if COM_RC_IN_MODE=2 if I remember correctly. Then whether strange things happen or not, depends on if you have Flight Mode config setup up for these channels. This is pretty easy to hit if you just switched your normal config to a HIL config.

You've changed the semantics of what manualControlReservedButtonCount is supposed to mean. It should match what the firmware supports. Firmware supports converting 8 buttons to rc channels as I pointed to. So for PX4 flight stack, those first 8 buttons get sent to the firmware. This is the same way the old joystick code worked. By setting reserved count to 0, you've just turned off that feature.

@LorenzMeier That said I don't quite get the usefulness of sending buttons to firmware. My suggestion is to remove this whole reversed count concept completely. And stop sending any button bits to MANUAL_CONTROL no matter what. But that is up to Lorenz since this was existing functionality that I"m maintaining.